scripts/helpers/openai-compatible-agent.py

```
@@ -110,12 +293,14 @@ def tool_exec(cwd: Path, name: str, args: dict) -> str:
                 return (f"ERROR: refused — {blocked}. This agent may not run that. "
                         f"Work within the project directory, or ask the operator to run it.")
             timeout = env_float("OPENAI_COMPAT_COMMAND_TIMEOUT", 20.0)
-            r = subprocess.run(cmd, cwd=str(cwd), shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, timeout=timeout, check=False)  # noqa: S602 - intentional shell-command tool
-            return (f"exit={r.returncode}\n" + r.stdout)[-20000:]
+            returncode, output = run_bounded_process(cmd, cwd, timeout, shell=True, output_limit=20000)
```

Ruff (0.16.3)

[error] Function call with `shell=True` parameter identified, security issue

(S604)